components/admin/requests/processing/TasksCard.tsx

```diff
@@ -90,20 +93,50 @@ export default function ProcessingTasksCard({ applicationId }: ProcessingTasksCa
     refetch();
   };
 
-  // TODO: Hook up to S3 upload
   const [uploadDocuments] =
     useMutation<UploadDocumentsResponse, UploadDocumentsRequest>(UPLOAD_DOCUMENTS_MUTATION);
-  const handleUploadDocuments = async (documentsUrl: string) => {
-    await uploadDocuments({ variables: { input: { applicationId, documentsUrl } } });
-    refetch();
-  };
 
   const [mailOut] = useMutation<MailOutResponse, MailOutRequest>(MAIL_OUT_APP_MUTATION);
   const handleMailOut = async (appMailed: boolean) => {
     await mailOut({ variables: { input: { applicationId, appMailed } } });
     refetch();
   };
 
+  const toast = useToast();
+  const [applicationDocuments, setApplicationDocuments] = useState<File | null>(null);
+
+  const handleSubmitDocuments = async (applicationDoc: File) => {
+    setApplicationDocuments(applicationDoc);
+    let documentS3ObjectKey = '';
+    if (applicationDoc) {
+      try {
+        const { key } = await clientUploadToS3(applicationDoc, 'rcd/application-documents');
+        documentS3ObjectKey = key;
+      } catch (err) {
+        toast({
+          status: 'error',
+          description: `Failed to upload documents: ${err}`,
+          isClosable: true,
+        });
+        setApplicationDocuments(null);
+        return;
+      }
+    }
+
+    await uploadDocuments({
+      variables: { input: { applicationId, documentsUrl: documentS3ObjectKey } },
+    });
+    refetch();
+  };
+
+  const handleUndoDocumentsUpload = async () => {
+    setApplicationDocuments(null);
+    await uploadDocuments({
+      variables: { input: { applicationId, documentsUrl: null } },
```

Comment:
Since we won't be saving documentsUrl in the db
```suggestion
      variables: { input: { applicationId, documentsS3ObjectKey: null } },
```